src/main/resources/db/migration/V1.2__Update_Beacon_Registration_Models.sql

```diff
@@ -0,0 +1,110 @@
+-- Update beacon tables to reflect correct models
+ALTER TABLE beacons
+    RENAME TO beacon;
+
+ALTER TABLE beacon
+    RENAME COLUMN battery_expiry TO battery_expiry_date;
+
+ALTER TABLE beacon
+    RENAME COLUMN last_serviced TO last_serviced_date;
+
+ALTER TABLE beacon
+    RENAME COLUMN serial_number TO manufacturer_serial_number;
+
+CREATE INDEX beacon_hex_id_index ON beacon(hex_id);
+
+ALTER TABLE beacon
+    ALTER COLUMN beacon_type SET NOT NULL,
+    ALTER COLUMN manufacturer SET NOT NULL,
+    ALTER COLUMN model SET NOT NULL,
+    ALTER COLUMN hex_id SET NOT NULL,
+    ALTER COLUMN manufacturer_serial_number SET NOT NULL,
+    ADD COLUMN chk_code text,
+    ALTER COLUMN beacon_status SET DEFAULT 'NEW',
+    DROP COLUMN checksum,
+    DROP COLUMN coding,
+    DROP COLUMN last_modified_date,
+    DROP COLUMN protocol_code;
+
+-- Update beacon use table
+ALTER TABLE beacon_uses
+    RENAME TO beacon_use;
+
+ALTER TABLE beacon_use
+    RENAME COLUMN use_type TO environment;
+
+ALTER TABLE beacon_use
+    ALTER COLUMN environment SET NOT NULL,
+    ADD COLUMN other_environment_use text,
+    ADD COLUMN activity text NOT NULL,
+    ADD COLUMN other_activity text,
+    ADD COLUMN call_sign text,
+    ADD COLUMN vhf_radio boolean,
+    ADD COLUMN fixed_vhf_radio boolean,
+    ADD COLUMN fixed_vhf_radio_input text,
+    ADD COLUMN portable_vhf_radio boolean,
+    ADD COLUMN portable_vhf_radio_input text,
+    ADD COLUMN satellite_telephone boolean,
+    ADD COLUMN satellite_telephone_input text,
+    ADD COLUMN mobile_telephone boolean,
+    ADD COLUMN mobile_telephone_input_1 text,
+    ADD COLUMN mobile_telephone_input_2 text,
+    ADD COLUMN purpose text,
+    ADD COLUMN max_capacity text,
+    ADD COLUMN vessel_name text,
+    ADD COLUMN homeport text,
+    ADD COLUMN area_of_operation text,
+    ADD COLUMN beacon_location text,
+    ADD COLUMN imo_number text,
+    ADD COLUMN ssr_number text,
+    ADD COLUMN official_number text,
+    ADD COLUMN rig_platform_location text,
+    ADD COLUMN aircraft_manufacturer text,
+    ADD COLUMN principal_airport text,
+    ADD COLUMN secondary_airport text,
+    ADD COLUMN registration_mark text,
+    ADD COLUMN hex_address text,
+    ADD COLUMN cn_or_msn_number text,
+    ADD COLUMN dongle boolean,
+    ADD COLUMN working_remotely_location text,
+    ADD COLUMN working_remotely_people_count text,
+    ADD COLUMN windfarm_location text,
+    ADD COLUMN windfarm_people_count text,
+    ADD COLUMN other_activity_location text,
+    ADD COLUMN other_activity_people_count text,
+    ADD COLUMN more_details text NOT NULL,
+    DROP COLUMN last_modified_date,
+    DROP COLUMN beacon_person_id,
+    DROP COLUMN vessel_id;
```

Comment:
> Captures all beacon use information (for now). We may want to split this out in the future dependent upon requirements like: capturing vessel/aircraft information per user so they can reference an aircraft/vessel they have created previously. As we do not currently have the functionality to create a vessel/aircraft per user, I have removed the Vessel model and db tables in this PR

Thinking about this again last night...  We have three options:
1.  Wide table, as in this PR.
2.  Normalize further, i.e. split out columns to `vessel`, `aircraft`, `other_use`.  (We discussed this yesterday and decided "not yet").
3.  Schema-on-write.  What about using the Postgres JSON type to store use-specific data? So `beacon_uses` table would look something like this:

```
| id | environment | purpose | activity | activity_details |
```

And the `activity_details` column would contain JSON set by the application.  If a vessel, this would be `vessel_name`, `homeport` etc., if an aircraft, `dongle`, `hex_address` etc.

Why would we do this?  I'm looking at the columns like `windfarm_location` above and wondering "What happens if there are future uses we haven't anticipated?"  If we use options 2 or 3, the MCA has to migrate a production DB to handle a new use.  If it's stored in JSON, only the application has to be extended.  One downside is lookup performance on, say, vessel IMO number.

What do you think @matthew-a-carr?  Wondering how schema-on-write for the beacon use might compare to the wide table option.

Reply:
@zackads Have gone back and forth on this this morning. Putting this in a [jsonb column](https://www.postgresql.org/docs/9.4/datatype-json.html) would allow us to have a _schema less_ column for additional beacon activity information but I feel even though this change is large, we should hopefully have the majority of the fields captured at this point and if we do need to add additional fields then the changes shouldn't be as wholesale as this one.